File: src/main/java/ink/ptms/sandalphon/module/api/NMS.java
package ink.ptms.sandalphon.module.api;

import io.izzel.taboolib.module.inject.TInject;
import org.bukkit.block.Block;
import org.bukkit.entity.Player;

/**
 * @Author sky
 * @Since 2020-05-30 17:00
 */
public abstract class NMS {

    @TInject(asm = "ink.ptms.sandalphon.module.api.NMSHandle")
    public static final NMS HANDLE = null;

    abstract public void sendBlockAction(Player player, Block block, int a, int b);

    abstract public void setBlockData(Block block, byte data);

}

File: src/main/java/ink/ptms/sandalphon/module/api/NMS.kt
package ink.ptms.sandalphon.module.api

import io.izzel.taboolib.module.inject.TInject
import org.bukkit.block.Block
import org.bukkit.entity.Player

/**
 * @author sky
 * @since 2020-05-30 17:00
 */
abstract class NMS {
    abstract fun sendBlockAction(player: Player?, block: Block?, a: Int, b: Int)
    abstract fun setBlockData(block: Block?, data: Byte)

    companion object {
        @TInject(asm = "ink.ptms.sandalphon.module.api.NMSHandle")
        val HANDLE: NMS? = null
    }
}